### PR TITLE
Add encryption implementation

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

File: src/encryption.rs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        // Read `eph_pub`
+        let mut eph_pub = [0u8; EPH_KEY_SIZE];
+        buffer.read_from_back(&mut eph_pub)?;
+        let eph_pub = Point::from_bytes(&eph_pub).map_err(|_| Error)?;
+
+        // Read `chacha_nonce`
+        let mut chacha_nonce = ChaChaNonce::default();
+        buffer.read_from_back(&mut chacha_nonce)?;
+
+        // Derive a `chacha_key` from `ehp_key`
```
Comment (jkatzDfns) [inline src/encryption.rs]: typo: ehp_key

**Review by jkatzDfns — DISMISSED** [review]

File: src/encryption.rs (hunk 1 above)
Comment (survived) [reply src/encryption.rs]:
```suggestion
        // Derive a `chacha_key` from `eph_key`
```

**Review by maurges — DISMISSED** [review]

File: src/encryption.rs [hunk 2]
```diff
@@ -37,19 +37,17 @@ impl DecryptionKey {
         buffer.read_from_back(&mut eph_pub)?;
         let eph_pub = Point::from_bytes(&eph_pub).map_err(|_| Error)?;
 
-        // Read `chacha_nonce`
-        let mut chacha_nonce = ChaChaNonce::default();
-        buffer.read_from_back(&mut chacha_nonce)?;
-
         // Derive a `chacha_key` from `eph_key`
-        let mut chacha_key = ChaChaKey::default();
+        let mut chacha_key = AesKey::default();
```
Comment (maurges) [inline src/encryption.rs]: Might want to change the names of variables to aes from chacha as well

**Review by maurges — APPROVED** [review]

File: src/encryption.rs (hunk 2 above)
Comment (survived) [reply src/encryption.rs]: Oops

**Pull request merged**